repos/system_upgrade/el7toel8/actors/mariadbcheck/libraries/library.py

```diff
@@ -0,0 +1,227 @@
+        'metadata_lock_info.so', 'mypluglib.so', 'mysql_clear_password.so',
+        'qa_auth_client.so', 'qa_auth_interface.so', 'qa_auth_server.so',
+        'query_cache_info.so', 'query_response_time.so', 'remote_io.so',
+        'server_audit.so', 'sha256_password.so', 'simple_password_check.so',
+        'sql_errlog.so', 'test_versioning.so', 'wsrep_info.so',
+        'daemon_example.ini' # TODO what with ini
+    ]
+    if not isdir(plugin_dir):
+        # TODO cant check plugins
+        return []
+    plugins_installed = listdir(plugin_dir)
+
+    return [plugin for plugin in plugins_installed if plugin not in plugins_in8]
+
+
+def get_warn_options(used_opts):
+    """ 
+        Return tuple(warn_changed[], warn_renamed{from: to}, warn_removed[]) of config options somehow changed in rhel8
+        https://mariadb.com/kb/en/library/upgrading-from-mariadb-55-to-mariadb-100/#options-that-have-been-removed-or-renamed 
+    """
+
+    options_changed_def = [
+        'aria-sort-buffer-size', 'back_log', 'innodb-buffer-pool-instances',
+        'innodb-concurrency-tickets', 'innodb-log-file-size',
+        'innodb-old-blocks-time', 'innodb-open-files', 
+        'innodb-purge-batch-size', 'innodb-undo-logs',
+        'max-connect-errors', 'max-relay-log-size', 'myisam-sort-buffer-size',
+        'optimizer-switch'
+    ]
+    options_renamed = {
+        'engine-condition-pushdown': 'set optimizer_switch="engine_condition_pushdown=on"',
+        'innodb-fast-checksum': 'innodb-checksum-algorithm',
+        'innodb-flush-neighbor-pages': 'innodb-flush-neighbors',
+        'innodb-stats-auto-update': 'innodb-stats-auto-recalc'
+    }
+    options_not_in8 = [
+        'innodb-adaptive-flushing-method', 'innodb-autoextend-increment',
+        'innodb-blocking-buffer-pool-restore', 'innodb-buffer-pool-pages',
+        'innodb-buffer-pool-pages-blob', 'innodb-buffer-pool-pages-index',
+        'innodb-buffer-pool-restore-at-startup', 'innodb-buffer-pool-shm-checksum',
+        'innodb-buffer-pool-shm-key', 'innodb-checkpoint-age-target',
+        'innodb-dict-size-limit', 'innodb-doublewrite-file',
+        'innodb-ibuf-accel-rate', 'innodb-ibuf-active-contract',
+        'innodb-ibuf-max-size', 'innodb-import-table-from-xtrabackup',
+        'innodb-index-stats', 'innodb-lazy-drop-table', 
+        'innodb-merge-sort-block-size', 'innodb-persistent-stats-root-page',
+        'innodb-read-ahead', 'innodb-recovery-stats',
+        'innodb-recovery-update-relay-log', 'innodb-stats-update-need-lock',
+        'innodb-sys-stats', 'innodb-table-stats', 'innodb-thread-concurrency-timer-based',
+        'innodb-use-sys-stats-table', 'xtradb-admin-command'
+    ]
+    
+    warn_changed = []
+    warn_renamed = {}
+    warn_removed = []
+
+    # TODO this could check also values
+    for changed_opt in options_changed_def:
+        for used_opt in used_opts:
+            if re.search(r"" + changed_opt + "[\s=]{1}", used_opt):
```

Comment:
Please use string formatting instead of concatenation
`re.search(r"%s[\s=]{1}" % changed_opt, used_opt)`